diff/diff.go

```diff
@@ -100,8 +105,42 @@ func WithLabels(labels map[string]string) Opt {
 
 // WithPayloads sets the apply processor payloads to the config
 func WithPayloads(payloads map[string]*types.Any) ApplyOpt {
-	return func(_ context.Context, _ ocispec.Descriptor, c *ApplyConfig) error {
+	return func(_ context.Context, _ *ocispec.Descriptor, c *ApplyConfig) error {
 		c.ProcessorPayloads = payloads
 		return nil
 	}
 }
+
+// WithApplyOptOciDescriptorLabels allows setting ocispec.Descriptor annotations
+func WithApplyOptOciDescriptorLabels(labels map[string]string) ApplyOpt {
```

Comment:
A couple of other options here:
* reuse `ApplyConfig.ProcesorPayloads`, which should be fine as long as we don't have a processor with the same ID as the key, which is probably not going to happen. This however, still feels like a hack (similar to the current implementation)
* introduce `ApplyConfig.Labels`, which to me seems to be a desirable approach? (not sure of the scope for cmifs, but it could also potentially use labels?) with this, we will have to update the diff service prootbufs

any thoughts on this? @kevpar @ambarve @katiewasnothere @dcantah

Reply:
What's the background on what the ProcessorPayloads are meant to be used for (if we know 😆)?

Reply:
from my understanding, layer content can be uncompressed/compressed or in some other custom format. for the first two it's handled with either simple write or gzip decompression accordingly, but for the custom format there's a support to start a binary. so these ProcessorPayloads are then marshalled via a named pipe to the binary process